doc/mgmt/Sonic_mirror.md

```diff
@@ -0,0 +1,356 @@
+It also listens to the FDB entry change (one entry update) for destination MAC getting added on a member port (for vlan with ip) and LAG member addition ( for a lag with ip) to set correct destination MAC and
+destination port in SAI if the VLAN and LAG have destination IP as a neighbor. Once a mirror session is created it's parameters cannot be altered.
+
+```
+mirror session mirror1
+
+   list of configurable parameters based on backend Support
+```
+
+#### 3.6.2.2 Show Commands
+
+#### show mirror sessions
+`
+show monitor session
+`
+
+```
+sonic(config)# show monitor session
+
+mirror session mir1
+source-ip 1.1.1.1
+destination-ip 2.2.2.2
+ip-dscp 20
+ip-ttl 10
+gre-protocol 0x6558
+queue 0
+status  active
+```
+#### show mirror-access-group  sessions
+
+`
+ show ip mirror-access-group
+`
+
+```
+sonic(config)# show ip mirror-access-group
+Ingress IP mirror-access-list acl_mirror on Ethernet116
+Ingress IP mirror-access-list acl_mirror on Ethernet112
+```
+#### show mirror-access-list
+
+`
+  show ip  mirror-access-list
+`
+
+```
+sonic(config)# do show ip mirror-access-lists
+ip mirror-access-list acl_mirror
+     10 permit udp 0.0.0.0/0 0.0.0.0/0 dscp 10 session mir1
+```
+
+#### Create a access control list for mirror session to filter packets based on acl rules before mirroring.
+
+`
+ip mirror-access-list <name>  -------this creates entry of type mirror in ACL table
+`
+
+```
+sonic(config)# ip mirror-access-list al-mirror
+```
+
+#### Create ACL rules for the mirror access list and associate a mirror session name to it.
+This create ACL rules in ACL rule table and mirrors only the packet that are not filtered by the ACL rules.
+The session id is required for mirror ip access-list.
+
+`
+seq <number> permit <other allowed ipv4 filters> capture <mirror session name>
+`
+
+
+
+```
+sonic(config)# ip mirror-access-list al-mirror
+sonic(config-ipv4-mirror-acl)# seq 2 permit tcp any src-eq 4096 any syn capture mir1
+```
+
+
+#### Delete ACL rules for mirror access list
+`
+no <sequence no>
+`
+```
+sonic(config)# ip mirror-access-list al-mirror
+sonic(config-ipv4-mirror-acl)# no seq 2
+```
+
+#### Delete a mirror ACL table and its rules
+
+`
+no ip mirror-access-list <mirror-access-list-name>
+`
+
+```
+sonic(config)# no ip mirror-access-list al-mirror
+```
+
+#### Associate the mirror-access list with an interface
+
+`
+ip mirror-access-group <name>
+`
+
+```
+sonic(config)# interface Ethernet 116
+sonic(conf-if-Ethernet116)# ip mirror-access-group al-mirror
+```
+
+#### Delete a mirror-access list from an interface
+
+`
+no ip mirror-access-group <mirror-access-list-name>
+`
+
+```
+sonic(config)# interface Ethernet 116
+sonic(conf-if-Ethernet116)# no ip mirror-access-group al-mirror
+```
+
+
+#### 3.6.2.3 Debug Commands
+#### 3.6.2.4 IS-CLI Compliance
+This CLI is not a ISL command and is based on what sonic supports.

```

Comment:
There is no port-mirroring support on Sonic currently. The CISCO or OS10 CLI is based on source and destination port-mirroring support for ersapn /local OR rspan.  Moreover as I have mentioned in the doc that the backed doesn't support  altering the tunnel parameters  once the session is created. So the proposed CLI is based on the current support in Sonic for session creation (only).
If broadcom is adding the support for port mirroring and session parameter alteration after creation then we can go ahead and discuss changing the proposed CLI.

Reply:
I believe port mirroring support will be added by rupesh soon. Its best if we keep the CLIs future proof to make sure when the features are supported in future, the CLIs dont to change drastically.

Reply:
The CLI's parameter may have more option as the back end support is added to make it configurable and we can update the HLD accordingly.  The syntax will remain more or less the same except ability to configure some new parameters.
Will the port mirroring support get added by altering any existing tables  ....like will the "Mirror_session_table" get augmented with  source or destination ports to support ERSPAN? Or is there a plan to add a new Config DB table for supporting ERSPAN. I am asking this because the current "Mirror_session_table" has parameters overlapping with some ERSPAN config parameters.  Also what's about the local and RSPAN in terms of config DB tables.

Reply:
There are 2 aspects we need to consider for this
1) How the CLIs will look like. This should be based in User experience and if there are enhancements needed, we need to do it as we cant change CLIs later.
2) How the Config DB schema will look like for SPAN, RSPAN and ERSPAN.

For the CLIs its preferable to have a mode change with just monitor session Name and by type if desired. Once the mode changes, User can configure different parameters for it. This will give flexibility to add/update individual parameters later. There can be 1 Line CLI to configure everything as long as mode changes is also supported as described earlier.

Example:
(config)# monitor session MyERSPAN type erspan-source
(config-monitor-session)# source ip X.X.X.X
(config-monitor-session)# destination ip X.X.X.X

On Config DB schema, will get back with specifics or will ask Rupesh to provide the details on the same.